src/app/shared/components/button/button.component.html

```diff
@@ -1,18 +1,27 @@
 <ng-container [ngSwitch]="type">
-  <a *ngSwitchCase="'link'" routerLink="[]" role="button" draggable="false" class="govuk-button" data-module="govuk-button">
+  <a
+    *ngSwitchCase="'link'"
+    role="button"
+    class="govuk-button"
+    data-module="govuk-button"
+    draggable="false"
+    [routerLink]="routerLink"
+    [id]="id"
+    [ngClass]="design ? 'govuk-button--' + design : ''"
```

Comment:
Is `govuk-button--` a valid class?

Reply:
Ignore that one

Reply:
`'govuk-button--' + design` is. If design is falsy then no class is added.